src/components/controllers/tutorial-page.jsx

```diff
@@ -0,0 +1,44 @@
+import { useRoute, useLocation } from 'preact-iso';
+import { SolutionProvider } from './tutorial/contexts';
+import { useContent } from '../../lib/use-resource';
+import { useTitle, useDescription } from './utils';
+import { NotFound } from './not-found';
+import { useLanguage } from '../../lib/i18n';
+import { tutorialRoutes } from '../../lib/route-utils';
+import { Tutorial } from './tutorial';
+
+export default function TutorialPage() {
+	const { params } = useRoute();
+	const { step } = params;
+
+	if (!tutorialRoutes[`/tutorial${step ? `/${step}` : ''}`]) {
+		return <NotFound />;
+	}
+
+	return <TutorialLayout />;
+}
+
+function TutorialLayout() {
+	const { url } = useLocation();
+	const { params } = useRoute();
+	const [lang] = useLanguage();
+
+	const { html, meta } = useContent([lang, !params.step ? 'tutorial/index' : url]);
+	useTitle(meta.title);
+	useDescription(meta.description);
+
+	// Preload the next chapter
+	// TODO: Webpack creates a circular dependency that
+	// it cannot resolve. Temporarily disabled
+	//useEffect(() => {
+	//	if (meta && meta.next) {
+	//		getContent([lang, meta.next]);
+	//	}
+	//}, [meta && meta.next, url]);
```

Comment:
This is frustrating, but Webpack will not cooperate and I can't figure out a workaround. I'm not even sure what's causing a circular dependency there.